components/backup/src/endpoint.rs

```diff
@@ -505,7 +505,7 @@ impl BackupRange {
     async fn backup_raw_kv_to_file<E: Engine>(
         &self,
         engine: E,
-        db: Arc<DB>,
+        db: RocksEngine,
```

Comment:
why it's generic on Engine, but still need to pass a rocksdb?

Reply:
/cc @3pointer

Reply:
I merge it now, and solve issues further if there is any.

Reply:
We need build `RocksSstWriterBuilder` from `RocksEngine` during backup. but it seems we cannot get `RocksEngine` from generic Engine 🤔

Reply:
maybe you don't need to call `set_db` for `RocksSstWriterBuilder`

Reply:
if we don't call  `set_db`,  it seems we cannot get `cf_options` of current rocksdb. this makes backup sst different with normal sst files.